### DISCUSSION TITLE
3D Audio

### DISCUSSION THREAD

**Discussion by FakeException** [AlmasB/FXGL#1082] [Q&A] [answered]
Is there a 3D Audio feature for 3D games?

**Comment by AlmasB** [comment, marked as answer]
Unfortunately, no. The underlying JavaFX API does not provide this functionality. There is only 2D positional audio, which makes use of volume and balance to shift the audio as needed.

I'd be interested to explore this topic further. I suppose the first step towards this support would be to check if there are any good 3D audio libs available in Java. If not, perhaps it is possible to provide bindings to OpenAL.

**Reply by FakeException** [reply]
thanks, it would be cool to have this feature, if you find something for doing it, please add it.
